File: backend/Turistando/Turistando.WebApi/Models/Usuario.cs
using System;
using System.Collections.Generic;
using Turistando.WebApi.Helpers;

namespace Turistando.WebApi.Models
{
    public class Usuario
    {
        public Usuario()
        {
            this.Roteiros = new List<Roteiro>();
            this.ReservasTurista = new List<Reserva>();
            this.ReservasGuia = new List<Reserva>();
        }

        public int Id { get; set; }
        public EnumTipoUsuario Tipo { get; set; }
        public string Nome { get; set; }
        public string Avatar { get; set; }
        public DateTime DataNascimento { get; set; }
        public string Sexo { get; set; }
        public string Cidade { get; set; }
        public string Estado { get; set; }
        public string Telefone { get; set; }
        public string Email { get; set; }
        public string Senha { get; set; }
        public string GuiaCarro { get; set; }
        public string GuiaPlaca { get; set; }
        public string GuiaHorario { get; set; }
        public string GuiaDetalhe { get; set; }
        public double GuiaAvaliacao { get; set; }

        public virtual ICollection<Roteiro> Roteiros { get; set; }
        public virtual ICollection<Reserva> ReservasTurista { get; set; }
        public virtual ICollection<Reserva> ReservasGuia { get; set; }
    }
}

File: frontend/turistando-react/src/models/usuario.ts
import {EnumTipoUsuario} from "../helpers/enum-tipo-usuario";
import {Roteiro} from "./roteiro";
import {Reserva} from "./reserva";

export class Usuario
{
    constructor()
    {
        this.Id = 0;
        this.Tipo = EnumTipoUsuario.Turista;
        this.Nome = '';
        this.Avatar = 'no-image.jpg';
        this.DataNascimento = new Date();
        this.Sexo = '';
        this.Cidade = '';
        this.Estado = '';
        this.Telefone = '';
        this.Email = '';
        this.Senha = '';
        this.GuiaCarro = '';
        this.GuiaPlaca = '';
        this.GuiaHorario = '';
        this.GuiaDetalhe = '';
        this.GuiaAvaliacao = 0;
        
        this.Roteiros = new Array<Roteiro>();
        this.ReservasTurista = new Array<Reserva>();
        this.ReservasGuia = new Array<Reserva>();
    }

    public Id: number;
    public Tipo: EnumTipoUsuario;
    public Nome: string;
    public Avatar: string;
    public DataNascimento: Date | Date[];
    public Sexo: string;
    public Cidade: string;
    public Estado: string;
    public Telefone: string;
    public Email: string;
    public Senha: string;
    public GuiaCarro: string;
    public GuiaPlaca: string;
    public GuiaHorario: string;
    public GuiaDetalhe: string;
    public GuiaAvaliacao: number;

    public Roteiros: Roteiro[];
    public ReservasTurista: Reserva[];
    public ReservasGuia: Reserva[];
}
